modules/oe_translation_cdt/src/Compiler/CdtParametersCompilerPass.php

```diff
@@ -9,20 +9,22 @@
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 
 /**
- * Registers the cdt.client_configuration container parameter.
+ * Registers the container parameters used by the CDT.
  */
 class CdtParametersCompilerPass implements CompilerPassInterface {
 
   /**
    * {@inheritdoc}
    */
   public function process(ContainerBuilder $container): void {
-    $container->setParameter('cdt.client_configuration', [
-      'apiBaseUrl' => Settings::get('cdt.base_api_url'),
-      'username' => Settings::get('cdt.username'),
-      'password' => Settings::get('cdt.password'),
-      'client' => Settings::get('cdt.client'),
-    ]);
+    // The container parameters can't be added as a single array.
+    // Array values don't support the "%" character, that may
+    // appear in the password or in the URL.
+    // @see https://www.drupal.org/project/drupal/issues/3458344
+    $container->setParameter('cdt.base_api_url', Settings::get('cdt.base_api_url'));
+    $container->setParameter('cdt.username', Settings::get('cdt.username'));
+    $container->setParameter('cdt.password', Settings::get('cdt.password'));
```

Comment:
when flushing the caches I get `The parameter "cdt.password" has a dependency on a non-existent parameter "ssword". Did you mean this: "cdt.password"?`
Symfony treats this `%` as a special character in the ParameterBag, so we need to double it, see ParameterBagL234, I think a simple `str_replace("%", "%%", Settings::get('cdt.password'));` here will do the trick

Reply:
Very weird, I will doublecheck!
The problem is that Drupal tries to do this `str_replace` by itself, so it may result with `%%%%`

Reply:
I still get the error, reinstalled everything from scratch, even re-pulled the docker containers after deleting them
This is the config with issues:
```
      CDT_BASE_API_URL: "https://example.com/api"
      CDT_CLIENT: "test_client"
      CDT_USERNAME: "test_username"
      CDT_PASSWORD: "tes#t_pa%ssw!or%%d"
      CDT_API_KEY: "test_apikey"
```
this is the command that generates the error
`./vendor/bin/drush en -y oe_translation_cdt_mock`

Reply:
Indeed, there was another escaping issue here. I've just added `str_replace`, and made an additional test to be 100% sure